src/main/kotlin/com/h0tk3y/player/MusicApp.kt

```diff
@@ -22,26 +28,62 @@ open class MusicApp(
         if (isClosed) return
         isClosed = true
 
-        /** TODO: Сохранить состояние плагинов с помощью [MusicPlugin.persist]. */
+        for (plugin in plugins) {
+            val stateFile = File(plugin.pluginId)
+            stateFile.createNewFile()
+            stateFile.outputStream().use {
+                plugin.persist(it)
+            }
+        }
     }
 
     fun wipePersistedPluginData() {
-        // TODO: Удалить сохранённое состояние плагинов.
+        for (plugin in plugins) {
+            val stateFile = File(plugin.pluginId)
+            if (stateFile.exists()) stateFile.delete()
+        }
     }
 
-    private val pluginClassLoader: ClassLoader = TODO("Создать загрузчик классов для плагинов.")
+    private val pluginClassLoader: ClassLoader =
+        URLClassLoader(pluginClasspath.map { it.toURI().toURL() }.toTypedArray())
 
     private val plugins: List<MusicPlugin> by lazy {
-        /**
-         * TODO используя [pluginClassLoader] и следуя контракту [MusicPlugin],
-         *      загрузить плагины, перечисленные в [enabledPluginClasses].
-         *      Эта функция не должна вызывать [MusicPlugin.init]
-         */
-        emptyList<MusicPlugin>()
+        enabledPluginClasses.map { plugin ->
+            val rawPlugin = try {
+                pluginClassLoader.loadClass(plugin)
+            } catch (e: ClassNotFoundException) {
+                throw PluginClassNotFoundException(plugin)
+            }
+            val primaryConstructor = rawPlugin.kotlin.primaryConstructor
+            if (primaryConstructor != null && primaryConstructor.parameters.size == 1 &&
+                primaryConstructor.parameters[0].type.classifier == MusicApp::class
+            ) {
+                primaryConstructor.call(this) as MusicPlugin
+            } else {
+                val constructor = rawPlugin.kotlin.constructors.singleOrNull {
+                    it.parameters.isEmpty()
+                }
+
+                val newPlugin = constructor?.call() as MusicPlugin
```

Comment:
If `constructor` is `null` here your code will throw wrong exception (`NullPointerException` instead of `IllegalPluginException`) at line 74.
You can fix this by adding `?: throw IllegalPluginException(rawPlugin)`